lib/relations.js

```diff
@@ -75,8 +75,7 @@ function RelationMixin() {
  * @property {Object} model Model object
  */
 RelationMixin.hasMany = function hasMany(modelTo, params) {
-  var def = RelationDefinition.hasMany(this, modelTo, params);
-  this.dataSource.adapter.resolve(this);
+  let def = RelationDefinition.hasMany(this, modelTo, params);
```

Comment:
Use `const` - see http://loopback.io/doc/en/contrib/style-guide.html#variable-declarations
  
Please fix other places in this file too.